### ISSUE TITLE
Bugfix: Reconcile references to `Group` and `GroupLevel`.

### ISSUE THREAD

**Issue by samussiah** [Gilead-BioStats/gsm#1592]
## Expected Behavior
Reference the hierarchy of the analysis (site/country/study/etc.) with a single term.

## Current Behavior
`Group` is found in the workflows and widgets; `GroupLevel` is found in the input/transform/analyze/flag/summarize functions.

## Steps to Reproduce

1.
2.
3.
4.

**Comment by jwildfire** [comment]
We should use `GroupLevel` throughout.

**Comment by jonthegeek** [comment]
We're also inconsistent about the values of these:

```r
unique(sampleGroups$GroupLevel)
#> [1] "Site"    "Study"   "Country"

unique(sampleInput$GroupLevel)
#> [1] "Site"

unique(sampleMetrics$GroupLevel)
#> [1] "site"

unique(sampleResults$GroupLevel)
#> [1] "siteid"

unique(sampleSummary$GroupLevel)
#> [1] "siteid"

MakeWorkflowList(strPath = system.file('workflow', 'metrics', package = 'gsm')) %>% 
  purrr::map(~.x$meta$GroupLevel) %>% 
  unique() %>% 
  unlist()
#> [1] "country" "site"
```

"Site" vs "site" vs "siteid", etc.

Should this be fixed in #1627, or as a separate ticket?
